core/include/usFrameworkEvent.h

```diff
@@ -41,13 +40,14 @@ class FrameworkEventData;
 /**
  * \ingroup MicroServices
  *
- * A general event from the Framework.
- *
+ * An event from the Micro Services framework describing a Framework event.
  * <p>
- * FrameworkEvent objects are delivered to {@code FrameworkListener}s
- * when a general event occurs within the CppMicroServices environment.
+ * <code>FrameworkEvent</code> objects are delivered to listeners connected
+ * via BundleContext::AddFrameworkListener() when an event occurs within the
+ * Framework which a user would be interested in. A <code>Type</code> code 
+ * is used to identify the event type for future extendability.
  *
- * @see FrameworkListener
+ * @see BundleContext#AddFrameworkListener
  */
 class US_Core_EXPORT FrameworkEvent
```

Comment:
I suggest to modify this class such that it works similar to `BundleEvent` and `Service Event`.  This would mean using a `std::shared_ptr` instead of the `SharedDataPointer` template (this will allow us to rely on the compiler generated destructor). Also remove the copy constructor and copy assignment operator.